services/analyzer/sequelize-tables-analyzer.js

```diff
@@ -141,86 +163,83 @@ async function analyzeTable(table, config) {
 
 // NOTICE: Use the foreign key and reference properties to determine the associations
 //         and push them as references of the table.
-function defineAssociationType(databaseSchema) {
-  Object.values(databaseSchema).forEach((table) => {
+function createAllReferences(databaseSchema, schemaGenerated) {
+  const references = {};
+  Object.keys(databaseSchema).forEach((tableName) => { references[tableName] = []; });
+
+  Object.keys(databaseSchema).forEach((tableName) => {
+    const table = databaseSchema[tableName];
     const { constraints, primaryKeys } = table;
-    constraints.forEach((constraint) => {
+    const { isJunction } = schemaGenerated[tableName].options;
+
+    const foreignKeysWithExistingTable = constraints
+      .filter((constraint) => constraint.columnType === 'FOREIGN KEY'
+        && databaseSchema[constraint.foreignTableName]);
```

Comment:
Is it a fix/protection compared to the current implementation?

Reply:
It's a protection and a fix a possible issues on cross schema. It should be in another PR I forgot to move it. Do you still want me to move it ?

Reply:
No but you should add a change description in the changelog if you think this was an issue in the previous implementation.